.gitlab-ci.yml

```diff
@@ -21,66 +33,88 @@ include:
     file: '.gitlab-ci-check-license.yml'
   - project: 'Northern.tech/Mender/mendertesting'
     file: '.gitlab-ci-github-status-updates.yml'
+  - project: 'Northern.tech/Mender/mendertesting'
+    file: '.gitlab-ci-check-docker-build.yml'
 
-test:unit:
-  stage: test
-  before_script:
-    - apt-get -qq update && apt-get install -yq $(cat deb-requirements.txt)
-    # start the dbus service
-    - service dbus start
+build:docker:
+  rules:
+    - when: never
+  script: ["true"]
 
-build:make:
-  stage: build
-  needs: []
-  image: golang:1.17-alpine3.14
-  before_script:
-    - apk add --update git make gcc pkgconfig libc-dev glib-dev
-  script:
-    - make build
+publish:docker:
+  rules:
+    - when: never
+  script: ["true"]
 
-generate-qa-trigger:
-  image: python:alpine
-  stage: trigger_prep
+publish:image:mender:
   rules:
-    - if: '$CI_COMMIT_BRANCH =~ /^(master|[0-9]+\.[0-9]+\.x)$/'
-    - if: '$CI_PIPELINE_SOURCE == "pipeline"'
-      when: never
-    - if: '$CI_PIPELINE_SOURCE == "schedule"'
-      when: never
-  before_script:
-    - apk add --no-cache git
-    - pip3 install pyyaml
-    - wget -q https://raw.githubusercontent.com/mendersoftware/mender-qa/master/scripts/generate_client_publish_job.py
-  script:
-    - python generate_client_publish_job.py --trigger ${CI_PROJECT_NAME} --version ${CI_COMMIT_REF_NAME} --filename gitlab-ci-client-qemu-publish-job.yml
-  artifacts:
-    paths:
-      - gitlab-ci-client-qemu-publish-job.yml
+    - when: never
+  script: ["true"]
+
+publish:image:saas:
+  rules:
+    - when: never
+  script: ["true"]
+
+trigger:saas:sync-staging-component:
+  rules:
+    - when: never
+  script: ["true"]
+
 
-trigger:mender-qa:
-  stage: trigger
+build:docker-multiplatform:
   rules:
-    - if: '$CI_COMMIT_BRANCH =~ /^(master|[0-9]+\.[0-9]+\.x)$/'
-    # the following is to prevent an endless loop of qa pipelines caused by downstream pipelines
-    - if: '$CI_PIPELINE_SOURCE == "pipeline"'
-      when: never
-    - if: '$CI_PIPELINE_SOURCE == "schedule"'
-      when: never
-  trigger:
-    include:
-      - artifact: gitlab-ci-client-qemu-publish-job.yml
-        job: generate-qa-trigger
+    - if: '$CI_COMMIT_REF_NAME =~ /^(main|staging)$/'
+      when: always
+    - when: manual
 
-trigger:integration:
-  stage: trigger
-  inherit:
-    variables: false
+publish:docker-multiplatform:
   rules:
-    - if: '$CI_COMMIT_BRANCH == "master"'
-    - if: '$CI_PIPELINE_SOURCE == "pipeline"'
-      when: never
-    - if: '$CI_PIPELINE_SOURCE == "schedule"'
-      when: never
-  trigger:
-    project: Northern.tech/Mender/integration
-    branch: master
+    - if: '$CI_COMMIT_REF_NAME =~ /^(main|staging)$/'
+      when: always
+    - if: '$CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+\.[0-9]+(-[0-9A-Za-z-]+)?/'
+      when: manual
+    - when: never
+  image: "quay.io/skopeo/stable:${SKOPEO_VERSION}"
```

Comment:
Good to me. Minor (optional) suggestion: switch `skopeo` for `regctl`, just to have a unique standard tool. reference: 

```
  image: "registry.gitlab.com/northern.tech/mender/mender-test-containers:docker-multiplatform-buildx-v1-master"

regctl image copy ${GITLAB_REGISTRY_TAG} $DOCKER_REPOSITORY:$DOCKER_PUBLISH_TAG
```
But up to you

Reply:
I prefer using as few in-house images as possible in our pipelines since this generally leads to more maintenance overhead.

Reply:
> I prefer using as few in-house images as possible in our pipelines since this generally leads to more maintenance overhead.

off-house images have given us headaches in the past too :sweat_smile: There it no golden rule.